src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -108,12 +95,54 @@ public ODataBatchReaderState State
                 return this.batchReaderState;
             }
 
-            private set
+            protected set
             {
                 this.batchReaderState = value;
             }
         }
 
+        /// <summary>
+        /// >The batch-specific URL converter that stores the content IDs found in a changeset and supports resolving cross-referencing URLs.
+        /// </summary>
+        internal ODataBatchPayloadUriConverter PayloadUriConverter

```

Comment:
>internal [](start = 8, length = 8)

Ideally, all access to the PayloadUriConverter would be in the base class, so we wouldn't need to expose this to the derived class.  Otherwise we would have to make this protected (not internal) since it shouldn't matter whether the derived class is in this assembly or a different assembly.  However, that would require making the ODataBatchPayloadUriConverter public, which I'd rather not do...

Reply:
duplicated comment? close this one first.
Please re-open otherwise.

---
In reply to: [141156150](https://github.com/OData/odata.net/pull/960#discussion_r141156150) [](ancestors = 141156150)